src/ip.rs

```diff
@@ -117,34 +117,18 @@ impl Ipv4Hdr {
     }
 }
 
-#[repr(C)]
-#[derive(Copy, Clone)]
-#[cfg_attr(feature = "serde", derive(::serde::Serialize, ::serde::Deserialize))]
-pub struct in6_addr {
-    pub in6_u: in6_u,
-}
-
-#[repr(C)]
-#[derive(Copy, Clone)]
-#[cfg_attr(feature = "serde", derive(::serde::Serialize, ::serde::Deserialize))]
-pub union in6_u {
-    pub u6_addr8: [u8; 16usize],
-    pub u6_addr16: [u16; 8usize],
-    pub u6_addr32: [u32; 4usize],
-}
```

Comment:
I didn't really see the point of keeping this union. IMO, you either wants the bytes in network-byte ordering or as `Ipv6Addr`. `[u16; 8]` and `[u32; 4]` are pretty pointless because the array itself would be big-endian ordered but the the individual numbers would be little-endian encoded on little-endian machines.

Reply:
I agree that the other array types are pointless for us. That's how the Linux kernel defines it and I think the major reason is that some drivers and subsystems (e.g. infiniband) make use of the `[u32; 4]` array:

https://elixir.bootlin.com/linux/v6.13.7/A/ident/u6_addr32

The `[u16; 8]` is not used anywhere though, lol:

https://elixir.bootlin.com/linux/v6.13.7/A/ident/u6_addr16

But anyways, we can stick to the byte arrays.